File: public/usage-examples/graphics/draw_ellipse/draw_ellipse-1-radiating-wave.py
from splashkit import *

open_window("Draw Ellipse", 800, 600)
clear_screen(color_white())

for i in range(30):
    
    width = 600 - i * 20
    height = 400 - i * 10
    x = 100 + i * 10
    y = 100 + i * 5

    random_color = rgb_color(
        rnd_int(255), rnd_int(255), rnd_int(255)   
    )

    draw_ellipse(random_color, x, y, width, height)

refresh_screen()
delay(4000)
close_all_windows()

File: public/usage-examples/graphics/draw_ellipse/draw_ellipse-1-radiating-wave.cpp
#include "splashkit.h"

int main()
{
    open_window("Draw Ellipse", 800, 600);
    clear_screen();

    // Draw 30 ellipses
    for (int i = 0; i < 30; i++)
    {   
        // Decrease width by 20 every round
        int width = 600 - i * 20;
        // Decrease height by 10 every round
        int height = 400 - i * 10;
        // Increase x position by 10 every round
        int x = 100 + i * 10;
        // Increase y position by 5 every round
        int y = 100 + i * 5;

        color randomColor = rgb_color(rnd(255), rnd(255), rnd(255));

        // Draw ellipse base on the given data
        draw_ellipse(randomColor, x, y, width, height);
    }

    refresh_screen();
    delay(4000);
    close_all_windows();

    return 0;
}
